src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(makeAppendString(F, builder, member.Name));
+                    block.Add(makeAppendString(F, builder, " = "));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
+                        F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendObject),
+                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+
+                    if (i < printableMembers.Length - 1)
+                    {
+                        block.Add(makeAppendString(F, builder, ", "));
+                    }
+                }
+
+                block.Add(F.Return(F.Literal(true)));
+
+                F.CloseMethod(F.Block(block.ToImmutableAndFree()));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                diagnostics.Add(ex.Diagnostic);
+                F.CloseMethod(F.ThrowNull());
+            }
+
+            static BoundStatement makeAppendString(SyntheticBoundNodeFactory F, BoundParameter builder, string value)
+            {
+                return F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendString), F.StringLiteral(value)));
+            }
+        }
+
+        internal static MethodSymbol? FindValidPrintMembersMethod(TypeSymbol containingType, CSharpCompilation compilation)
+        {
+            if (containingType.IsErrorType() || containingType.IsObjectType())
+            {
+                return null;
+            }
+
+            MethodSymbol? candidate = null;
+            var accessibility = (containingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && containingType.IsSealed)
+                ? Accessibility.Private
+                : Accessibility.Protected;
+
+            var stringBuilder = TypeWithAnnotations.Create(compilation.GetWellKnownType(WellKnownType.System_Text_StringBuilder));
+
+            foreach (var member in containingType.GetMembers(WellKnownMemberNames.PrintMembersMethodName))
+            {
+                if (member is MethodSymbol { DeclaredAccessibility: var foundAccessibility, IsStatic: false, ParameterCount: 1, Arity: 0, ParameterTypesWithAnnotations: var parameters } method &&
```

Comment:
>parameters [](start = 173, length = 10)

Consider naming "parameterTypes".  Or simply remove that part of the pattern and check `method.Parameters[0].Type` below, which would avoid creating an array for the parameter types.